include/real/const_precision_iterator.hpp

```diff
@@ -0,0 +1,391 @@
+#ifndef BOOST_CONST_PRECISION_ITERATOR_HPP
+#define BOOST_CONST_PRECISION_ITERATOR_HPP
+
+#include "real/boundary_helper.hpp"
+#include "real/interval.hpp"
+#include "real/real_explicit.hpp"
+#include "real/real_algorithm.hpp"
+#include "real/real_operation.hpp"
+#include "real/real_helpers.hpp"
+#include <memory>
+#include <variant>
+#include <assert.h>
+#include <iterator>
+
+namespace boost {
+    namespace real{
+    /**
+     * @file The const_precision_iterator provides the functionality to iterate through precision intervals
+     * of all three kinds of reals
+     * 
+     * @note variant and visit/visitors are used extensively in this implementation
+     * @TODO: consider doing traversals to lower the demands of recursion
+     * @sa documention on std::variant, std::visit
+     */
+        template<class... Ts> struct overloaded : Ts... { using Ts::operator()...; };
+        template<class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
+        // fwd decl
+        class real;
+        class operation_helpers;
+
+        class const_precision_iterator {
+            public:
+            /**
+             * @brief Determines the maximum precision to use
+             */
+            static std::optional<unsigned int> maximum_precision;
```

Comment:
This was in the master branch already. What do you mean by the second part?

Reply:
I mean that knowing the operands max precision you can compute the current number one. In case we talking about a number contrived as a binary operation. Otherwise, it should be defined in the number. Why is it global to the whole class?